code/cycler/src/wattrex_battery_cycler/mid/mid_str/mid_str_node.py

```diff
@@ -83,21 +86,34 @@ def __apply_command(self, command : MidStrCmdDataC) -> None:
         if command.cmd_type == MidStrReqCmdE.GET_NEW_EXP:
             log.info('Getting new experiment info from database')
             exp_info, battery_info, profile_info = self.db_iface.get_start_queued_exp()
-            self.__actual_exp_id = exp_info.exp_id
+            if exp_info is not None:
+                self.__actual_exp_id = exp_info.exp_id
+            ## If there is an error gathering experiment info, manager will manage it
             log.debug("Sending new experiment to APP_MANAGER")
             self.str_data.send_data(MidStrCmdDataC(cmd_type= MidStrDataCmdE.EXP_DATA,
                     experiment= exp_info, battery= battery_info, profile= profile_info))
         elif command.cmd_type == MidStrReqCmdE.GET_EXP_STATUS:
-            exp_status = self.db_iface.get_exp_status(exp_id= self.__actual_exp_id)
+            if self.__actual_exp_id == -1:
+                log.warning("No experiment is running")
+                exp_status = None
+            else:
+                exp_status = self.db_iface.get_exp_status(exp_id= self.__actual_exp_id)
             self.str_data.send_data(MidStrCmdDataC(cmd_type= MidStrDataCmdE.EXP_STATUS,
                                                    exp_status= exp_status))
         elif command.cmd_type == MidStrReqCmdE.GET_CS:
             cycler_info = self.db_iface.get_cycler_station_info()
             self.str_data.send_data(MidStrCmdDataC(cmd_type= MidStrDataCmdE.CS_DATA,
                                                    station= cycler_info))
         elif command.cmd_type == MidStrReqCmdE.SET_EXP_STATUS and command.exp_status is not None:
-            self.db_iface.modify_current_exp(exp_status= command.exp_status,
+            if self.__actual_exp_id == -1:
+                log.warning("No experiment is running")
+            else:
+                self.db_iface.modify_current_exp(exp_status= command.exp_status,
                                              exp_id= self.__actual_exp_id)
+        elif command.cmd_type == MidStrReqCmdE.TURN_DEPRECATED:
+            log.info("Turning cycler station to deprecated")
```

Comment:
If deprecate a CS we should stop the node or we will stop it with the flag condition from APP_MAN?

Reply:
The node should be stopped once all experiments are in error. After the end of the function, as in the closure of the node a commit should be done.

Reply:
Okey, implement this